pkg/controller/cluster/main-controller.go

```diff
@@ -375,7 +414,7 @@ func (c *Controller) Start(threadiness int, stopCh <-chan struct{}) error {
 			if isOperatorTLS() {
 				publicCertPath, publicKeyPath := c.generateOperatorTLSCert()
 				klog.Infof("Starting HTTPS API server")
-				close(apiServerWillStart)
+				wg.Done()
```

Comment:
a simple `defer wg.Done()` at the start of the goroutine would make it so you don't have to copy the code to every case, a panic con `c.generateOperatorTLSCert()` could cause the `wg.Done()` to never be called

Reply:
as per our conversation. we will address this with channels to handle the webserver exit as well in another PR